### PR TITLE
Configurable GPIO option ARCH_NR_GPIO for x86 architecture

### PR DESCRIPTION
**Pull request by msiddare**
The x86 platform did not allow configuring the maximum number of GPIOs
supported, although the ARM platform did. For cisco-8000 platform,
each FPGA gpio IP block can support 1K pins. Distributed chassis with
Route processor and Fabric card can have 10 such IP blocks, along with
additional pins through i2c gpio extenders.

This patch supports configurable number of GPIO's at kernel config time
similar to ARM platform.

Signed-off-by: Madhava Reddy Siddareddygari <msiddare@cisco.com>

### CODE REVIEW

File: patch/cisco-x86-gpio-config.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
++++ b/arch/x86/Kconfig
+@@ -1033,6 +1033,15 @@ config SCHED_MC_PRIO
+ 
+ 	  If unsure say Y here.
+ 
++config ARCH_NR_GPIO
++	int
++	prompt "Maximum number of GPIO's"
++	default 512
++	---help---
```
Comment (paulmenzel) [inline patch/cisco-x86-gpio-config.patch:32]: Why the dashes?

File: patch/cisco-x86-gpio-config.patch [hunk 2]
```diff
@@ -0,0 +1,42 @@
+From 2ee30f75a96d5f7841ffd58cd65410160302421b Mon Sep 17 00:00:00 2001
+From: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+Date: Tue, 13 Jul 2021 18:43:16 -0700
+Subject: [PATCH] Configurable GPIO option ARCH_NR_GPIO for x86 architecture
```
Comment (paulmenzel) [inline patch/cisco-x86-gpio-config.patch:4]: [Please make it a statement.](https://chris.beams.io/posts/git-commit/)

**Review by paulmenzel — COMMENTED** [review]
Please talk to upstream, if the Kconfig variable should be moved to a common place.

File: patch/cisco-x86-gpio-config.patch (hunk 2 above)
Comment (msiddare) [reply patch/cisco-x86-gpio-config.patch]: Updated patch description following example from kernel.org for similar change

File: patch/cisco-x86-gpio-config.patch (hunk 1 above)
Comment (msiddare) [reply patch/cisco-x86-gpio-config.patch]: Most of the config options [ Not all ] have convention of adding dashes to the "help".
Only very few do not have dashes.  

Patch updated to remove dashes

**Review by lguohan — APPROVED** [review]

File: patch/series [hunk 3]
```diff
@@ -84,10 +84,12 @@ net-sch_generic-fix-the-missing-new-qdisc-assignment.patch
 0030-hwmon-Add-convience-macro-to-define-simple-static-se.patch
 0031-backport-nvme-Add-hardware-monitoring-support.patch
 0032-platform-mellanox-mlxreg-hotplug-Use-capability-regi.patch
-
```
Comment (paulmenzel) [inline patch/series:87]: Why remove the blank line?

File: patch/cisco-x86-gpio-config.patch [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ 
++config ARCH_NR_GPIO
++	int
++	prompt "Maximum number of GPIO's"
++	default 512
++	help
++	  Maximum number of GPIOs in the system.
++
++	  If unsure, leave the default value.
++
```
Comment (paulmenzel) [inline patch/cisco-x86-gpio-config.patch:36]: As written you need to talk to upstream, if it’s fine to add the option to `arch/x86/Kconfig`, or if it should be merged with the ARM option and moved to some common location.

File: patch/series (hunk 3 above)
Comment (lguohan) [reply patch/series:87]: i did that to handle merge conflict.

**Pull request merged**